astroplan/scheduling.py

```diff
@@ -90,6 +90,80 @@ def from_exposures(cls, target, priority, time_per_exposure,
         return ob
 
 
+class Scorer(object):
+    """
+    Returns scores and score arrays from the evaluation of constraints on
+    observing blocks
+    """
+    def __init__(self, blocks, observer, schedule):
+        """
+
+        Parameters
+        ----------
+        blocks : list of `~astroplan.scheduling.ObservingBlock` objects
+            list of blocks that need to be scored
+        observer : `~astroplan.Observer`
+            the observer
+        schedule : `~astroplan.scheduling.Schedule`
+            The schedule inside which the blocks should fit
+        """
+        self.blocks = blocks
+        self.observer = observer
+        self.schedule = schedule
+        # the lists below should change if there are new constraints made
+        self.constraint_list = ['AltitudeConstraint', 'AirmassConstraint', 'AtNightConstraint',
```

Comment:
In our Google Hangout conversation today, we've decided that we can take a different approach: first loop over each block and append constraints that you find in each block to a list (if they're not already in the list). That way you don't have to know anything about what constraints have been implemented, which will naturally support user-defined constraints (which are really important!).